application/apps/indexer/sources/src/command/process.rs

```diff
@@ -188,6 +188,9 @@ impl ByteSource for ProcessSource {
             let stored = line.len() + 1;
             self.buffer.copy_from_slice(line.as_bytes());
             self.buffer.copy_from_slice(b"\n");
+            // BUG: This should represent all available bytes in `self.buffer`.
```

Comment:
also correct observation! please expose this bug with a unit test and then fix it